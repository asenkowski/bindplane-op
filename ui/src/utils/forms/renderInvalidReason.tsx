import { Box, Typography } from "@mui/material";

export function renderInvalidReason(reason: string): JSX.Element {
  // Split up the invalid reason by new line and map to typography elements.
  const lines = reason.split("\n");
  return (
    <Box marginTop={1}>
      {lines.map((l) => (
        <Typography fontSize={12} fontFamily="monospace" key={l} color="error">
          {l}
        </Typography>
      ))}
    </Box>
  );
}
