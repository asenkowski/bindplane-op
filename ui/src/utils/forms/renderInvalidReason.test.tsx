import { render } from "@testing-library/react";
import { renderInvalidReason } from "./renderInvalidReason";
import renderer from "react-test-renderer";

const REASON = `1 error occurred:\n* unable to parse spec.raw as yaml: yaml: line 9: could not find expected ':'`;

describe("renderInvalidReason", () => {
  it("renders without error", () => {
    render(<>{renderInvalidReason(REASON)}</>);
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<>{renderInvalidReason(REASON)}</>);

    expect(tree).toMatchSnapshot();
  });
});
