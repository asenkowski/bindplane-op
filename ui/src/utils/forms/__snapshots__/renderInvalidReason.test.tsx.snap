// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderInvalidReason matches snapshot 1`] = `
<div
  className="MuiBox-root css-164r41r"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1sxhpes-MuiTypography-root"
  >
    1 error occurred:
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1sxhpes-MuiTypography-root"
  >
    * unable to parse spec.raw as yaml: yaml: line 9: could not find expected ':'
  </p>
</div>
`;
